File: 70-climbing-stairs/70-climbing-stairs.py
class Solution:
    def climb(self, n, memo):
        if n in memo:
            return memo[n]
        
        memo[n] = self.climb(n - 1, memo) + self.climb(n - 2, memo)
        return memo[n]
        
    def climbStairs(self, n: int) -> int:
        memo = {
            1: 1,
            2: 2
        }
        
        return self.climb(n, memo)

File: 70-climbing-stairs/70-climbing-stairs.go
func find(steps int, memo map[int]int) int {
    if v, contains := memo[steps]; contains {
        return v
    }
    
    if steps == 0 {
        return 1
    }
    
    if steps < 0 {
        return 0
    }
    
    memo[steps - 1] = find(steps - 1, memo) 
    memo[steps - 2] = find(steps - 2, memo)
    return memo[steps - 1] + memo[steps - 2]
}

func climbStairs(n int) int {
    memo := make(map[int]int)
    return find(n, memo)
}